dlt/common/destination/reference.py

```diff
@@ -49,6 +51,39 @@ def __init__(
         ) -> None:
             ...
 
+@configspec(init=True)
+class DestinationClientStagingConfiguration(DestinationClientDwhConfiguration):
+    as_staging: bool = False
+    bucket_url: str
+    credentials: Union[AwsCredentialsPure, GcpCredentials]
+
+    if TYPE_CHECKING:
+        def __init__(
+            self,
+            destination_name: str = None,
+            credentials: Union[AwsCredentialsPure, GcpCredentials] = None,
+            dataset_name: str = None,
+            default_schema_name: Optional[str] = None,
+            as_staging: bool = False,
+            bucket_url: str = None,
+        ) -> None:
+            ...
+
+@configspec(init=True)
+class DestinationClientDwhWithStagingConfiguration(DestinationClientDwhConfiguration):
```

Comment:
OK I like this idea! so let's double down on it:
1. seems you do not need the full config just staging credentials!
2. put those credentials to `DestinationClientDwhConfiguration` as `staging_credentials`, use generic credentials type (see the `credentials` in base class)
3. another big +1: user can fill those credentials explicitly (it will not work fully now - we will deal with that later)
4. we do not need DestinationClientDwhWithStagingConfiguration